code/modules/vehicles/entered.dm

```diff
@@ -35,22 +35,74 @@
 	if(!istype(M))
 		return FALSE
 	if(!silent)
-		M.visible_message("<span class='boldnotice'>[M] climbs into \the [src]!</span>")
+		M.visible_message("<span class='notice'>[M] climbs into \the [src]!</span>")
 	M.forceMove(src)
 	add_occupant(M)
 	return TRUE
 
-/obj/vehicle/sealed/proc/mob_try_exit(mob/M, mob/user, silent = FALSE)
-	mob_exit(M, silent)
+/obj/vehicle/sealed/proc/mob_try_exit(mob/M, mob/user, silent = FALSE, randomstep = FALSE)
+	mob_exit(M, silent, randomstep)
 
-/obj/vehicle/sealed/proc/mob_exit(mob/M, silent = FALSE)
+/obj/vehicle/sealed/proc/mob_exit(mob/M, silent = FALSE, randomstep = FALSE)
 	if(!istype(M))
 		return FALSE
 	remove_occupant(M)
 	M.forceMove(exit_location(M))
+	if(randomstep)
+		var/turf/target_turf = get_step(exit_location(M), pick(GLOB.cardinals))
+		M.throw_at(target_turf, 5, 10)
+		
 	if(!silent)
-		M.visible_message("<span class='boldnotice'>[M] drops out of \the [src]!</span>")
+		M.visible_message("<span class='notice'>[M] drops out of \the [src]!</span>")
 	return TRUE
 
 /obj/vehicle/sealed/proc/exit_location(M)
 	return drop_location()
+
+/obj/vehicle/sealed/attackby(obj/item/I, mob/user, params)
+	if(key_type && !is_key(inserted_key) && is_key(I))
+		if(user.transferItemToLoc(I, src))
+			to_chat(user, "<span class='notice'>You insert \the [I] into \the [src].</span>")
+			if(inserted_key)	//just in case there's an invalid key
+				inserted_key.forceMove(drop_location())
+			inserted_key = I
+		else
+			to_chat(user, "<span class='notice'>[I] seems to be stuck to your hand!</span>")
+		return
+	return ..()
+
+/obj/vehicle/sealed/proc/remove_key(mob/user)
+	if(!inserted_key)
+		return
+		to_chat(user, "<span class='notice'>There is no key in \the [src]!</span>")
+	if(!is_occupant(user) || !(occupants[user] & VEHICLE_CONTROL_DRIVE))
+		to_chat(user, "<span class='notice'>You must be driving \the [src] to remove [src]'s key!</span>")
+		return
+	to_chat(user, "<span class='notice'>You remove \the [inserted_key] from \the [src].</span>")
+	inserted_key.forceMove(drop_location())
+	user.put_in_hands(inserted_key)
+	inserted_key = null
+	return TRUE
+
+/obj/vehicle/sealed/deconstruct(disassembled = TRUE)
+	. = ..()
+	DumpMobs()
+	explosion(src.loc, 0, 1, 2, 3, 0)
```

Comment:
src. is not needed and why are you making every vehicle explode by default?

Reply:
because its extremely funny